manifests/prepare.pp

```diff
@@ -16,11 +16,40 @@
     ensure => directory,
   }
 
-  file { $::puppet_agent::params::ssldir:
-    ensure  => directory,
-    source  => $::puppet_ssldir,
-    backup  => false,
-    recurse => true,
+  $ssl_dir = $::puppet_agent::params::ssldir
```

Comment:
I also think it does make the lines below where we append to that path slightly more readable